src/fromager/commands/graph.py

```diff
@@ -153,6 +162,69 @@ def explain_duplicates(wkctx, graph_file):
             print(f"  * No single version of {dep_name} meets all requirements")
 
 
+@graph.command()
+@click.option(
+    "--version",
+    type=clickext.PackageVersion(),
+    multiple=True,
+    help="filter by version for the given package",
+)
+@click.option(
+    "--recursive-depth",
+    type=int,
+    default=0,
+    help="recursively get why each package depends on each other. Set depth to -1 for full recursion till root",
+)
+@click.option(
+    "--req-type",
+    type=clickext.RequirementType(),
+    multiple=True,
+    help="filter by req type",
```

Comment:
We use "req" as an abbreviation a lot internally, but in the user interface maybe we should spell out "requirement".